lib/oli/delivery/sections.ex

```diff
@@ -1383,6 +1389,76 @@ defmodule Oli.Delivery.Sections do
     resource_type_id == container
   end
 
+  # Function that generates a set of unique slugs that don't collide with any of the existing ones from the
+  # section_resources table. It aims to minimize the number of queries to the database for ensuring that the slugs
+  # generated are unique.
+  defp generate_slugs_until_uniq(published_resources) do
+    # generate initial slugs for new section resources
+    published_resources_by_slug =
+      Enum.reduce(published_resources, %{}, fn {_, pr}, acc ->
+        title = pr.revision.title
+
+        # if a previous published resource has the same revision then generate a new initial slug different from the default
+        slug_attempt = if Map.has_key?(acc, Slug.slugify(title)), do: 1, else: 0
+        initial_slug = Slug.generate_nth(title, slug_attempt)
+
+        Map.put(acc, initial_slug, pr)
+      end)
+
+    # until all slugs are unique or up to 10 attempts
```

Comment:
Should we raise or something if we achieve the 10th attempt and we still have non unique slugs? With current impl we are returning all of them even though we could have records in `prs_by_non_uniq_slug`, or I misunderstood something?